server/datastore/mysql/software_titles.go

```diff
@@ -37,12 +37,12 @@ SELECT
 	COUNT(si.id) as software_installers_count,
 	COUNT(vat.adam_id) as vpp_apps_count
 FROM software_titles st
-LEFT JOIN software_titles_host_counts sthc ON sthc.software_title_id = st.id
+LEFT JOIN software_titles_host_counts sthc ON sthc.software_title_id = st.id AND (%s)
 LEFT JOIN software_installers si ON si.title_id = st.id AND si.global_or_team_id = ?
 LEFT JOIN vpp_apps vap ON vap.title_id = st.id
 LEFT JOIN vpp_apps_teams vat ON vat.global_or_team_id = ? AND vat.adam_id = vap.adam_id AND vat.platform = vap.platform
 WHERE st.id = ? AND
-	((sthc.hosts_count > 0 AND %s) OR vat.adam_id IS NOT NULL OR si.id IS NOT NULL)
```

Comment:
Do we need to put `platform` in the WHERE statement? I thought I had a switch statement I was using somewhere, but I can't find it.

Reply:
Something like this but in SQL? 
![image](https://github.com/user-attachments/assets/e7c66eb4-c294-4e78-919e-3fcf9f584af2)

Reply:
No need, because this is SoftwareTitleByID (so we don't need to filter the platform/source, it's already being filtered by ID)